x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -166,12 +166,23 @@ func createCloudWatchEvents(getMetricDataResults []cloudwatch.MetricDataResult,
 
 	// AWS EC2 Metrics
 	mapOfMetricSetFieldResults := make(map[string]interface{})
-	for _, output := range getMetricDataResults {
-		if len(output.Values) == 0 {
-			continue
+
+	// Find a timestamp for all metrics in output
+	timestamp := aws.FindTimestamp(getMetricDataResults)
+	if !timestamp.IsZero() {
+		for _, output := range getMetricDataResults {
```

Comment:
In the worst case, we iterated 3 timest through this array. We should check if we can simplify this in the future.

Reply:
Yeah good point, I will add this to my todo list.  Will track it with https://github.com/elastic/beats/issues/11213